libphobos/src/std/math.d

```diff
@@ -4456,61 +4470,58 @@ private:
 private:
     static uint getIeeeFlags()
     {
-        version(D_InlineAsm_X86)
-        {
-            asm pure nothrow @nogc
-            {
-                 fstsw AX;
-                 // NOTE: If compiler supports SSE2, need to OR the result with
-                 // the SSE2 status register.
-                 // Clear all irrelevant bits
-                 and EAX, 0x03D;
-            }
-        }
-        else version(D_InlineAsm_X86_64)
+        version(InlineAsm_X86_Any)
         {
-            asm pure nothrow @nogc
+            ushort sw;
+            asm pure nothrow @nogc { fstsw sw; }
+
+            // OR the result with the SSE2 status register (MXCSR).
+            if (haveSSE)
             {
-                 fstsw AX;
-                 // NOTE: If compiler supports SSE2, need to OR the result with
-                 // the SSE2 status register.
-                 // Clear all irrelevant bits
-                 and RAX, 0x03D;
+                uint mxcsr;
+                asm pure nothrow @nogc { stmxcsr mxcsr; }
+                return (sw | mxcsr) & EXCEPTIONS_MASK;
             }
+            else return sw & EXCEPTIONS_MASK;
         }
         else version (GNU)
         {
-            uint result;
-            version (X86)
+            version (X86_Any)
             {
+                ushort sw;
                 asm pure nothrow @nogc
                 {
-                    "fstsw %%ax; andl $0x03D, %%eax;" : "=a" result;
+                    "fstsw %0" : "=a" (sw);
                 }
-            }
-            else version (X86_64)
-            {
-                asm pure nothrow @nogc
+                // OR the result with the SSE2 status register (MXCSR).
+                if (haveSSE)
                 {
-                    "fstsw %%ax; andq $0x03D, %%rax;" : "=a" result;
+                    uint mxcsr;
+                    asm pure nothrow @nogc
+                    {
+                        "stmxcsr %0" : "=m" (mxcsr);
```

Comment:
Without looking, it should bw guarded by haveSSE. Which uses core.cpuid for x86.